great_expectations/expectations/expectation.py

```diff
@@ -1340,36 +1340,30 @@ def _validate(
         total_count = metrics.get("table.row_count")
         null_count = metrics.get("column_values.nonnull.unexpected_count")
         unexpected_count = metrics.get(self.map_metric + ".unexpected_count")
+        unexpected_values = metrics.get(self.map_metric + ".unexpected_values")
+        unexpected_index_list = metrics.get(self.map_metric + ".unexpected_index_list")
 
-        success = None
         if total_count is None or null_count is None:
+            total_count = nonnull_count = 0
+        else:
+            nonnull_count = total_count - null_count
+
+        success = None
+        if total_count == 0 or nonnull_count == 0:
             # Vacuously true
             success = True
```

Comment:
I don't think we should mix `bool` with `None` here. Can we use: 

```
success = total_count == 0 or nonnull_count == 0
``` 

so the default value is `True`/`False`. If we need to change the value later (i.e. L1357, we can do so).

Reply:
@cdkini For now, I want to keep this part intact for `_format_map_output()` to work as before (could be for another PR).